mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -299,51 +329,57 @@ case class ScalaPresentationCompiler(
       val pos = sourcePosition(driver, params, uri)
       val trees = driver.openedTrees(uri)
       val path = Interactive.pathTo(trees, pos)
-      val tp = Interactive.enclosingType(trees, pos)
+      val enclosing = expandRangeToEnclosingApply(path, pos)
+      val tp = enclosing.head.tpe
       val tpw = tp.widenTermRefExpr
 
       if tp.isError || tpw == NoType || tpw.isError then ju.Optional.empty()
       else
-        Interactive.enclosingSourceSymbols(path, pos) match
+        Interactive.enclosingSourceSymbols(enclosing, pos) match
           case Nil =>
             ju.Optional.empty()
-          case symbols =>
+          case symbols @ (symbol :: _) =>
             val printer = SymbolPrinter()
             val docComments =
               symbols.flatMap(ParsedComment.docOf(_))
-            val hoverString = symbols.headOption.map { symbol =>
+            val history = driver.compilationUnits.get(uri) match
+              case Some(unit) =>
+                val newctx =
+                  ctx.fresh.setCompilationUnit(unit)
+                val context =
+                  MetalsInteractive.contextOfPath(enclosing)(using newctx)
+                new ShortenedNames(IndexedContext(context))
+              case None => new ShortenedNames(IndexedContext(ctx))
+            val hoverString =
               tpw match
                 // https://github.com/lampepfl/dotty/issues/8891
                 case tpw: ImportType =>
-                  val history = new ShortenedNames(IndexedContext(ctx))
                   printer.hoverDetailString(
                     symbol,
                     history,
                     symbol.paramRef
                   )
                 case _ =>
-                  driver.compilationUnits.get(uri) match
-                    case Some(unit) =>
-                      val newctx =
-                        ctx.fresh.setCompilationUnit(unit)
-                      val tpdPath = Interactive.pathTo(
-                        newctx.compilationUnit.tpdTree,
-                        pos.span
-                      )(using newctx)
-                      val context =
-                        MetalsInteractive.contextOfPath(tpdPath)(using newctx)
-                      val history = new ShortenedNames(IndexedContext(context))
-                      printer.hoverDetailString(symbol, history, tpw)(using
-                        context
-                      )
-                    case None =>
-                      val history = new ShortenedNames(IndexedContext(ctx))
-                      printer.hoverDetailString(symbol, history, tpw)
-            }
-            val content = hoverContent(hoverString, docComments)
-            ju.Optional.of(new Hover(content))
+                  printer.hoverDetailString(symbol, history, tpw)
```

Comment:
```suggestion
                  printer.hoverDetailString(symbol, history, symbol.info)
```
Here it's needed to put tpe of symbols instead of expression to render symbol signature correctly (without precise resulting type). And probably the changes in `HoverMarkup` won't be required.

Reply:
This seems to break some other tests. I could just use type generated from `path` for signature, what do you think?

`HoverMarkup` change will still be needed as we would get a different typ for signature and expression.

Reply:
I reverted the changes to the expressions, so this PR should only influence Scala 3 expression types.

Reply:
I started looking what is wrong with just `symbol.info`.

The main difference was in this sample:
```scala
Lis@@t(1)
// Scala2 symbol signature: def apply[A](a: A*): List[A]
// Scala3 symbol signature: def apply[A](a: A*): CC[A]
```

Then I take a look at `HoverProvider` in Scala2 and realized that we haven't ported [`seenFromType`](https://github.com/scalameta/metals/blob/db3ae6638a8c16093aa3de5cb06c8e2e1a8b4f0d/mtags/src/main/scala-2/scala/meta/internal/pc/HoverProvider.scala#L135-L151) logic.

I added a kind of `seenFromType` + reverted the changes from HoverMarkup in [this wip commit](https://github.com/dos65/metals/commit/20cb4e805bb7cba2ce78ad42b24dcc70abd6e6e4) to validate if it's a key and got more close result to Scala2.

There are still failed tests but I think they are valid and we need to adjust object/class/trait .. rendering.
Could you look this commit? 
Honestly, it's in half-working state so I you want I can complete it a bit later.